lib/key/utils.js

```diff
@@ -101,3 +104,67 @@ export async function cloneKey(inputKey) {
     const [key] = await getKeys(inputKey.toPacketlist().write());
     return key;
 }
+
+/**
+ * Generate ECDHE key and secret from public key
+ *
+ * @param {Array<Options>}                              Public key Q, fingerprint and curve (name or OID)
+ * @returns {Promise<{V: Uint8Array, Z: Uint8Array}>}   Returns public part of ephemeral key and generated ephemeral secret
+ * @async
+ */
+export async function genPublicEphemeralKey(options) {
+    const curveObj = new openpgp.crypto.publicKey.elliptic.Curve(options.Curve);
+    const oid = new openpgp.OID(curveObj.oid);
+    const { V, S } = await openpgp.crypto.publicKey.elliptic.ecdh.genPublicEphemeralKey(curveObj, options.Q);
+    const cipher_algo = openpgp.enums.read(openpgp.enums.symmetric, curveObj.cipher);
+
+    const param = openpgp.crypto.publicKey.elliptic.ecdh.buildEcdhParam(
+        openpgp.enums.publicKey.ecdh,
+        oid,
+        curveObj.cipher,
+        curveObj.hash,
+        options.Fingerprint
+    );
+
+    const Z = await openpgp.crypto.publicKey.elliptic.ecdh.kdf(
+        curveObj.hash,
+        S,
+        openpgp.crypto.cipher[cipher_algo].keySize,
+        param,
+        curveObj,
+        false
+    );
+
+    return {V, Z}
+}
+
+/**
+ * Generate ECDHE secret from private key and public part of ephemeral key
+ *
+ * @param {Array<Options>}         Private key d, public part of ECDHE V, fingerprint and curve (name or OID)
```

Comment:
Wrong type, it's an Object.
Can you use destructuring to improve the readability of options ? (_it's easier to doc as we can do per options + types + what we need without looking at the body_)